latest/index.bs

```diff
@@ -331,24 +317,917 @@ Conforming readers:
+  <tr><th>`mapAxis`
+    <td>`"mapAxis":Dict[String:String]`
+    <td> A `maxAxis` transformation specifies an axis permutation as a map between axis names.
+  <tr><th>`translation`
+    <td> one of: <br>`"translation":List[number]`, <br>`"path":str`
+    <td>translation vector, stored either as a list of numbers (`"translation"`) or as binary data at a location
+    in this container (`path`).
+  <tr><th>`scale`
+    <td> one of: <br>`"scale":List[number]`, <br>`"path":str`
+    <td>scale vector, stored either as a list of numbers (`scale`) or as binary data at a location in this
+    container (`path`).
+  <tr><th>`affine`
+    <td> one of: <br>`"affine":List[number]`, <br>`"path":str`
+    <td>affine transformation matrix stored as a flat array stored either with json uing the affine field
+    or as binary data at a location in this container (path). If both are present, the binary values at path should be used.
+  <tr><th>`rotation`
+    <td> one of: <br>`"rotation":List[number]`, <br>`"path":str`
+    <td>rotation transformation matrix stored as an array stored either
+        with json or as binary data at a location in this container (path).
+        If both are present, the binary parameters at path are used.
+  <tr><th>`sequence`
+    <td> `"transformations":List[Transformation]`
+    <td>A sequence of transformations, Applying the sequence applies the composition of all transforms in the list, in order.
+  <tr><th>`displacements`
+    <td>`"url":str`<br>`"path":str`<br>`"interpolation":str`
+    <td>Displacement field transformation in this container (path) or another container located at (url).
+  <tr><th>`coordinates`
+    <td>`"url":str`<br>`"path":str`<br>`"interpolation":str`
+    <td>Coordinate field transformation in this container (path) or another container located at (url).
+  <tr><th>`inverseOf`
+    <td>`"transform":Transform`
+    <td>The inverse of a transformation. Useful if a transform is not closed-form invertible.  See Forward and inverse for details and examples.
+  <tr><th>`bijection`
+    <td>`"forward":Transform`<br>`"inverse":Transform`
+    <td>Explicitly define an invertible transformation by providing a forward transformation and its inverse.
+  <tr><th>`byDimension`
+    <td>`"transformations":List[Transformation]`
+    <td>Define a high dimensional transformation using lower dimensional transformations on subsets of
+    dimensions.
  <thead>
    <tr><th>type<th>fields<th>description
 </table>
 
-The transformations in the list are applied sequentially and in order.
+
+Conforming readers:
+- SHOULD be able to apply transformations to points
+- SHOULD be able to apply transformations to images
+
+Coordinate transformations from array to physical coordinates MUST be stored in multiscales ([[#multiscale-md]]),
+and MUST be duplicated in the atrributes of the zarr array. Transformations between different images MUST be stored in the
```

Comment:
If we are putting transformation metadata with the individual arrays, why do we need to duplicate it in group metadata?

Reply:
I assume the idea is to provide consolidated metadata and avoid reading many json files to find the relevant transformations. This at least has been the motivation to ONLY store this information in the group level metadata in v0.4. And from our experiences with object storage this is crucial for providing smooth data access. 
(E.g. for a multi-scale dataset with 10 scale arrays, we can read all transformations from one json (= accessing one remote file) rather than 11 (group level data + 1 json for each array)).

Reply:
> I'm surprised that loading a handful of JSON files is a performance burden -- if the fetches are done concurrently, the performance impact should be minimal, especially compared to the hundreds of thousands of data chunks that will also be fetched.

The problem is that the json fetches for the transformation data need to be done for initialization and cannot be done lazily (unlike fetching the data chunks). So this is blocking and significantly increases the start-up time.

> If the performance associated metadata access is the main concern, we should do consolidated metadata through a [dedicated mechanism](https://zarr.readthedocs.io/en/stable/tutorial.html#consolidating-metadata) rather than implement it on an attribute-by-attribute basis.

I agree in principle, but as far as I know this is not universally supported beyond python.

Reply:
> > I'm surprised that loading a handful of JSON files is a performance burden -- if the fetches are done concurrently, the performance impact should be minimal, especially compared to the hundreds of thousands of data chunks that will also be fetched.
> 
> The problem is that the json fetches for the transformation data need to be done for initialization and cannot be done lazily (unlike fetching the data chunks). So this is blocking and significantly increases the start-up time.

Could you hide the latency through parallelization? As far as I understand object storage, you can access as many objects as you want (up to 100s) at once and will incur the same latency as accessing 1 object (minus the actual download time, which is negligible for the small json files). Depending on many factors, the latency is on the order of 10s-100s ms.



> > If the performance associated metadata access is the main concern, we should do consolidated metadata through a [dedicated mechanism](https://zarr.readthedocs.io/en/stable/tutorial.html#consolidating-metadata) rather than implement it on an attribute-by-attribute basis.
> 
> I agree in principle, but as far as I know this is not universally supported beyond python.

I guess that is true. It is not part of the v2 or v3 spec. There are talks of adding it to v3 as an extension, though.